src/lib/autoinstall/storage_proposal.rb

```diff
@@ -0,0 +1,103 @@
+require "y2storage"
+
+module Y2Autoinstallation
+  # Storage proposal for AutoYaST
+  #
+  # This class it is mainly a wrapper around proposal Y2Storage proposal classes.
+  # Depending on the profile, it will select the right class: if 'partitioning'
+  # section is missing (or empty), {Y2Storage::GuidedProposal} will be used. Otherwise,
+  # proposal will be handdled by {Y2Storage::AutoinstProposal}.
+  class StorageProposal
+    include Yast::Logger
+
+    # @return [Y2Storage::GuidedProposal,Y2Storage::AutoinstProposal] Y2Storage proposal instance
```

Comment:
I'm starting to feel that we miss a common superclass here. But let's leave this for a future discussion.